src/classes/ADDR_Validator_TDTM.cls

```diff
@@ -68,21 +68,22 @@ public class ADDR_Validator_TDTM extends TDTM_Runnable {
             recordsPreviouslyProcessed = true;
         }
 
-        if(!recordsPreviouslyProcessed) {
-            Addr_Verification_Settings__c settings = UTIL_CustomSettingsFacade.getAddressVerificationSettings();
-            if ((triggerAction == TDTM_Runnable.Action.BeforeInsert || triggerAction == TDTM_Runnable.Action.BeforeUpdate) &&
-               settings.enable_automatic_verification__c) {
+        if(!recordsPreviouslyProcessed && UTIL_CustomSettingsFacade.getAddressVerificationSettings().enable_automatic_verification__c) {
+            if (triggerAction == TDTM_Runnable.Action.BeforeInsert || triggerAction == TDTM_Runnable.Action.BeforeUpdate) {
                 for(SObject so : newList) {
                     Address__c address = (Address__c)so;
+                    //don't verify this address if it has already been verified
+                    if (address.Verified__c != null && address.Verified__c) {
+                        continue;
+                    }
                     if(!system.isFuture() && !system.isBatch()) {
                         address.Verification_Status__c = Label.Addr_Pending_Verification;
                     } else {
                         address.Verification_Status__c = Label.Addr_Verification_Required;
                     }
                 }
             }
-            if ((triggerAction == TDTM_Runnable.Action.AfterInsert || triggerAction == TDTM_Runnable.Action.AfterUpdate) &&
-               settings.enable_automatic_verification__c) {
+            if (triggerAction == TDTM_Runnable.Action.AfterInsert || triggerAction == TDTM_Runnable.Action.AfterUpdate) {
```

Comment:
why do the afterInsert/Update handlers now process without checking enable_automatic_verification__c?

Reply:
@njjc still wondering about this one.  shouldn't it continue to check for enable-automatic_verification__c?

Reply:
It's still there in the first if statement... just refactored a bit.

Reply:
had to look at the full file.  looking at just the diff I missed it.  thanks.